sdk/identity/azure-identity/test/perf/inc/azure/identity/test/environment_credential_test.hpp

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * @file
+ * @brief Test the overhead of authenticating with secret credential.
+ *
+ */
+
+#pragma once
+
+#include <azure/identity.hpp>
+#include <azure/perf.hpp>
+
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace Azure { namespace Identity { namespace Test {
+
+  /**
+   * @brief A test to measure the authentication token performance.
+   *
+   */
+  class EnvironmentCredentialTest : public Azure::Perf::PerfTest {
```

Comment:
Is the goal here to compare the overhead of different `TokenCredentials` within C++ or compare it across languages?

Reply:
both

Reply:
And what are the results/findings from that comparison?

Reply:
well , for the moment we have no history to compare results to just the "previous version" that we choose based on pipeline settings.